Tests/kch fix failing on windows and stabilize

kamilchodola opened the pull request:
**Fixes** | Closes | Resolves 
No Issue number - adjusting tests on Windows and ubuntu - part1

## Changes:
- Added missing tests (5 projects). @flcl42 and @LukaszRozmej adjusted Db and KeyStore tests to pass.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [Y] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [X] No

rubo commented:
I'm not sure why this PR includes other not-yet merged PRs. If it should include, then we need to close those PRs.

rubo commented on .github/workflows/run-nethermind-tests.yml:
```diff
@@ -201,6 +201,42 @@ jobs:
+    name: Running Nethermind Tests 6
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Please update to `actions/checkout@v3`.

kamilchodola replied:
Done :) Thx!

rubo commented on .github/workflows/run-nethermind-tests.yml:
```diff
@@ -201,6 +201,42 @@ jobs:
+      with:
+        submodules: true
+    - name: Setting up dotnet
+      uses: actions/setup-dotnet@v1
```
Please update to `actions/setup-dotnet@v2`.

kamilchodola replied:
Done :) Thx!

kamilchodola commented:
@rubo @flcl42 @LukaszRozmej @MarekM25 @marcindsobczak 
Would ask for re-review.

Removed other changes from Alexey and Łukasz since those are already merged to master on other PRs and made small improvements as suggested by Ruben.
Additionally, I've added those 5 missing projects to codecoverage pipeline.
